EDEngineer/Utils/BlueprintFilters.cs

```diff
@@ -147,44 +156,90 @@ private void LoadSavedFilters()
 
         private void InsertMagicFilters()
         {
-            var magicTypeFilters = TypeFilter.MagicFilter;
-            TypeFilters.Insert(0, magicTypeFilters);
+            _magicTypeFilter = TypeFilter.MagicFilter;
+            TypeFilters.Insert(0, _magicTypeFilter);
 
-            var magicEngineerFilters = EngineerFilter.MagicFilter;
-            EngineerFilters.Insert(0, magicEngineerFilters);
+            _magicEngineerFilter = EngineerFilter.MagicFilter;
+            EngineerFilters.Insert(0, _magicEngineerFilter);
 
-            var magicGradeFilter = GradeFilter.MagicFilter;
-            GradeFilters.Insert(0, magicGradeFilter);
+            _magicGradeFilter = GradeFilter.MagicFilter;
+            GradeFilters.Insert(0, _magicGradeFilter);
 
-            var synthesisTypeFilter = new TypeFilter("@Synthesis", "TFSynthesis") { Checked = true };
+            var synthesisTypeFilter = new TypeFilter("@Synthesis", "TFSynthesis");
             TypeFilters.Add(synthesisTypeFilter);
 
-            var technologyTypeFilter = new TypeFilter("@Technology", "TFTechnology") { Checked = true };
+            var technologyTypeFilter = new TypeFilter("@Technology", "TFTechnology");
             TypeFilters.Add(technologyTypeFilter);
+        }
 
-            magicGradeFilter.PropertyChanged += (o, e) =>
+        private void UpdateFiltersFromMagicFilter(CollectionViewSource source, bool isChecked, IEnumerable<BlueprintFilter> filters)

```

Comment:
I'm confused with these two methods with the same name (`UpdateFiltersFromMagicFilter`)

could you eventually split that in a separate review - to make it more clear what bug/enhancement the change is addressing

Reply:
- I noticed that an engineer line "@merchant" appeared in engineers filter, I did the same as  "@Technology" and "@Synthesis".
- I noticed a lag while testing when checking "All types" filter, it's due to a cascade effect, each filter is activated/desactivated one by one, and a view refresh and all filters are executed each time, I changed that to refresh once. 
- UpdateFiltersFromMagicFilter is called when you check/uncheck an "all" filter to update children filters without recompute all. And UpdateMagicFilterFromFilters is an improvement : when you check/uncheck the last type/engineer/grade from the filter, it updates the "All" filter.
- I've also noticed a strange behaviour with @synthesis and @technology (and repro #573), i'ts was because the "magic filters" were inserted after the computation of "Allfilters" variable

Reply:
I see thanks!